ui/selectors/multichain/networks.ts

```diff
@@ -0,0 +1,141 @@
+
+import {
+  type ProviderConfigState,
+  type SelectedNetworkClientIdState,
+  getProviderConfig,
+  getNetworkConfigurationsByChainId,
+} from '../../../shared/modules/selectors/networks';
+import { createDeepEqualSelector } from '../../../shared/modules/selectors/util';
+import {
+  getIsBitcoinSupportEnabled,
+  getIsSolanaSupportEnabled,
+} from '../selectors';
+
+// Selector types
+
+export type MultichainNetworkControllerState = {
+  metamask: InternalMultichainNetworkState;
+};
+
+export type NetworkConfigurationsState = {
+  metamask: {
+    networkConfigurations: Record<
+      string,
+      InternalMultichainNetworkConfiguration
+    >;
+  };
+};
+
+export type SelectedNetworkChainIdState = {
+  metamask: Pick<
+    InternalMultichainNetworkState,
+    'selectedMultichainNetworkChainId'
+  >;
+};
+
+export type IsEvmSelectedState = {
+  metamask: Pick<InternalMultichainNetworkState, 'isEvmSelected'>;
+};
+
+export type MultichainNetworkConfigurationsByChainIdState = {
+  metamask: {
+    multichainNetworkConfigurationsByChainId: Record<
+      string,
+      InternalMultichainNetworkConfiguration
+    >;
+    networkConfigurationsByChainId: Record<
+      string,
+      InternalNetworkConfiguration
+    >;
+  };
+};
+
+/**
+ * This type takes into account the state
+ * of the multichain-network-controller and
+ * the network-controller.
+ */
+export type MultichainNetworkConfigState =
+  MultichainNetworkConfigurationsByChainIdState &
+    SelectedNetworkChainIdState &
+    IsEvmSelectedState &
+    SelectedNetworkClientIdState &
+    ProviderConfigState;
+
+// Selectors
+
+export const getNonEvmMultichainNetworkConfigurationsByChainId = (
+  state: MultichainNetworkConfigurationsByChainIdState,
+) => state.metamask.multichainNetworkConfigurationsByChainId;
+
+export const getMultichainNetworkConfigurationsByChainId =
+  createDeepEqualSelector(
+    getNonEvmMultichainNetworkConfigurationsByChainId,
+    getNetworkConfigurationsByChainId,
+    getIsBitcoinSupportEnabled,
+    getIsSolanaSupportEnabled,
+    (
+      nonEvmNetworkConfigurationsByChainId,
+      networkConfigurationsByChainId,
+      isBitcoinSupportEnabled,
+      isSolanaSupportEnabled,
+    ): Record<CaipChainId, InternalMultichainNetworkConfiguration> => {
+      const filteredNonEvmNetworkConfigurationsByChainId: Record<
+        CaipChainId,
+        InternalMultichainNetworkConfiguration
+      > = {};
+
+      // This is not ideal but since there are only two non EVM networks
+      // we can just filter them out based on the support enabled
+      if (isBitcoinSupportEnabled) {
+        filteredNonEvmNetworkConfigurationsByChainId[BtcScope.Mainnet] =
+          nonEvmNetworkConfigurationsByChainId[BtcScope.Mainnet];
+      }
+
+      if (isSolanaSupportEnabled) {
+        filteredNonEvmNetworkConfigurationsByChainId[SolScope.Mainnet] =
+          nonEvmNetworkConfigurationsByChainId[SolScope.Mainnet];
+      }
```

Comment:
Add conditional for non EVM accounts

Reply:
Done in https://github.com/MetaMask/metamask-extension/pull/30309/commits/23bcd6eb32936c81e77c49ea11e80ec4ed9b0e0b